gen/src/builtin.rs

```diff
@@ -350,11 +350,11 @@ pub(super) fn write(out: &mut OutFile) {
         writeln!(out, "template <>");
         writeln!(out, "class impl<Error> final {{");
         writeln!(out, "public:");
-        writeln!(out, "  static Error error(repr::PtrLen repr) noexcept {{");
+        writeln!(out, "  static void error(const char* msg, size_t len) {{");
         writeln!(out, "    Error error;");
-        writeln!(out, "    error.msg = static_cast<char const *>(repr.ptr);");
-        writeln!(out, "    error.len = repr.len;");
-        writeln!(out, "    return error;");
+        writeln!(out, "    error.msg = msg;");
```

Comment:
what's the lifetime/ownership of `msg`? Is it pointing to Rust-owned memory?

Reply:
I have added a comment to cxx.h (not sure what's the best place for it): it points to c++ memory and will belong to c++.

https://github.com/cloudflare/workerd-cxx/blob/master/src/result.rs#L60